[Communication] - Administration - Managed Identity for PhoneNumberClient

minnieliu opened the pull request:
Draft for now: Blocked on running the Live tests for Managed Identity support for TNM. Waiting for service team to enable this functionality and should be available after Jan 15th.

Sandido committed:
LogAnalytics Throttle export API new parameters GroupByClientApplicationId and GroupByUserAgent (#17769)

* throttle input update

* log analyticsinputbase

* update requestratebyIntervalInput.cs

* add new test and record

Co-authored-by: Theodore Chang <thchan@microsoft.com>

abatishchev committed:
Made parsing storage connection string settings accept all cases (make check case-insensitive) (#17642)

* Made dictionary for connection string elements case-insensitive

* Update StorageConnectionStringTests.cs

* Update StorageConnectionStringTests.cs

* Update StorageConnectionStringTests.cs

* Update StorageConnectionString.cs

* Update StorageConnectionStringTests.cs

* Update StorageConnectionStringTests.cs

* Update StorageConnectionString.cs

* Update StorageConnectionString.cs

* Added to validValues.Contains

* Update CHANGELOG.md

azure-sdk committed:
allow configuration of a suffix for each config item. (#17860)

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>

giakas committed:
[AMS] Fixed a assemblyfile version mismatch bug. (#17855)

Commit:
Add maintenance tests for SQL SDK (#17723)

azure-sdk committed:
Revert "Remove old Update-Change-Log.ps1 (#1301)" (#17864)

This reverts commit b967cb62bd8db451a2f02d903f3658cf21f92db1.

Co-authored-by: Chidozie Ononiwu <31145988+chidozieononiwu@users.noreply.github.com>

Commit:
Managed Identity for TNM

Commit:
Remove duplicate

Commit:
Update README

hivyas committed:
added new customization file and changed swagger link (#17863)

pakrym committed:
Prepare for Azure.Core 1.8.1 release (#17878)

Commit:
update readme

Commit:
Adding readme env variables detail

azure-sdk committed:
Fix condition for test generation in prepare-pipelines (#17881)

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>

JoshLove-msft committed:
Add AsSystemEventData (#17871)

Fixes https://github.com/Azure/azure-sdk-for-net/issues/17298

pakrym committed:
Update Azure.Core to 1.8.1 (#17886)

azure-sdk committed:
Increment version for core releases (#17891)

kasobol-msft committed:
[Storage] SAS Credential in Storage. (#17646)

* Add AzureSasCredential

* corner case.

* api.

* core as project ref (todo undo this)

* first client.

* hack azure core in webjobs for now.

* api.

* constructors.

* blob tests.

* datalake tests.

* share tests + take out sas on share and service clients as there isn't much APIs that would work with sas there.

* queues tests.

* well that works. nvm.

* remarks.

* error message.

* predicate shouldn't be optional.

* post-merge tweaks.

* message about right UriBuilder.

* added uri validation.

* changelog.

* user delegation sas change.

* batch.

* this test doesn't work well in playback mode.

* pr feedback.

* comments.

* validation.

* undo project references workaround.

* undo webjobs project ref workaround.

audunn committed:
[NetAppFiles]Anf 8257 net app files 2020 09 01 (#17697)

* feat: bump to api version 2020-09-01

* feat:Add test recordings

* update version number

Co-authored-by: Audunn Baldvinsson <audunn.baldvinsson@netapp.com>

JoshLove-msft committed:
Use AzureSasCredential (#17893)

pakrym committed:
Allow null offset in legacy checkpoints (#17896)

kinelski committed:
[MetricsAdvisor] Added tests.yml to MetricsAdvisor (#17894)

azure-sdk committed:
Add quotes around the parameter (#17889)

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>

JianpingChen commented on sdk/communication/Azure.Communication.Administration/tests/Infrastructure/PhoneNumberAdministrationClientTestEnvironment.cs:
```diff
@@ -8,7 +8,9 @@ namespace Azure.Communication.Administration.Tests
+        private const string EndpointStringEnvironmentVariableName = "COMMUNICATION_ENDPOINT_STRING";
 
         public string ConnectionString => GetRecordedVariable(ConnectionStringEnvironmentVariableName);
+        public string EndpointString => GetRecordedVariable(EndpointStringEnvironmentVariableName);
```
Ideally, we should only need one config variable. Since end point is contained in the connection string already, can we just parse it from the connection string instead of adding another env variable? It'll simplify env configuration.

minnieliu replied:
Hey JP, technically we could, but the Endpoint and Connection string are used for two different ways of authentication. Endpoint can be used for token credential where as connection string is used for access key authentication.

RezaJooyandeh replied:
Let's do the environment variable cleanup once we finish the `Administration` splitting refactor. I already break the build once 😑

JianpingChen commented on sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs:
```diff
@@ -31,6 +33,28 @@ public PhoneNumberAdministrationClientLiveTests(bool isAsync) : base(isAsync)
+            }
+            else
+            {
+                tokenCredential = new DefaultAzureCredential();
```
Should this be loaded from env config?

JianpingChen replied:
My stupidity. DefaultAzureCredential is loading from env config.

JianpingChen commented on sdk/communication/Azure.Communication.Administration/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:
```diff
@@ -24,6 +26,32 @@ public Sample2_PhoneNumberAdministrationClient(bool isAsync) : base(isAsync)
         }
 
+        [Test]
+        public async Task CreatePhoneNumberWithTokenCredential()
```
Is this really just "sample" or is it runnable test?

minnieliu replied:
It is a sample that can be used in our README.md but it is also runnable to ensure that the sample is always up to date with the latest code

JianpingChen commented on sdk/communication/Azure.Communication.Administration/src/PhoneNumberAdministrationClient.cs:
```diff
@@ -22,31 +22,47 @@ public class PhoneNumberAdministrationClient
 
-        internal PhoneNumberAdministrationClient(PhoneNumberAdministrationClientOptions options, ConnectionString connectionString)
+        private PhoneNumberAdministrationClient(PhoneNumberAdministrationClientOptions options, ConnectionString connectionString)
             : this(new ClientDiagnostics(options), options.BuildHttpPipeline(connectionString), connectionString.GetRequired("endpoint"))
```
It feels more natural to put important arg first, connectionString.

minnieliu replied:
This is done to avoid naming conflicts.

RezaJooyandeh replied:
this is a bit tricky, but I had to do it this way to avoid conflict with the public constructor with the same argument that cannot control nullability.

christothes committed:
add release date for 3.0.0-beta.5 (#17904)

azure-sdk committed:
Increment version for tables releases (#17905)

Increment package version after release of Azure.Data.Tables

jsquire committed:
[Event Hubs Client] Idempotent Publishing Tweaks (#17906)

# Summary

The focus of these changes is to allow querying partition publishing properties when idempotent publishing is not enabled, returning an empty set of data with the feature flags.  Several areas of documentation were also enhanced to provide additional details and context.

# Last Upstream Rebase

Tuesday, December 12, 8:19am (EST)

chamons committed:
[Synapse] Add LRO for Spark operations (#17739)

- Add a LRO for spark operations that require special Livy poling
- The logic in these LRO is ported from test helpers written by the client team (and removed here as well)
- This is a rebase of https://github.com/Azure/azure-sdk-for-net/pull/17677 
- I had to shut up a few tests due to more nullability issues (fixing in https://github.com/Azure/azure-rest-api-specs/pull/12258)

Co-authored-by: Mariana Rios Flores <mariari@microsoft.com>
Co-authored-by: Christopher Scott <chriscott@hotmail.com>

pakrym committed:
Check client instrumentation (#17640)

Adds a guard rail that verifies that at least one client was instrumented if the test has any recordings.

kinelski committed:
[MetricsAdvisor] Fixing tests.yml (#17910)

christothes committed:
Implement TableUriBuilder and TableSasBuilder.Parse (#17890)

* Implement TableUriBuilder and Uri parsing for TableSasBuilder / TableAccountSasBuilder

minnieliu committed:
[Communication] - ALL - Adding PowerShell Docs to README (#17895)

* Adding PowerShell to README

* Update wording

* Fix links

* Update wording

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

seanmcc-msft committed:
Prepared Storage for release (#17908)

Commit:
Update ADT control-plane SDK to 2020-12-01 (#17775)

DorothySun216 committed:
Update dependency of Microsoft.Azure.Amqp to version 2.4.9 (#17912)

* add changelog for a new nuget minor version bump

* Revert "add changelog for a new nuget minor version bump"

This reverts commit 7ece7b55ea4e6d7af11fcb63a21bb9dd39373e53.

* add change log for amqp lib update

heaths committed:
Make CorsOptions.maxAgeInSeconds x-nullable (#17900)

Fixes #16928

Mohit-Chakraborty committed:
Add perf tests for Blob storage scenarios. (#17897)

* Add perf tests for Blob storage scenarios.

Create stand-alone test apps for Azure.Storage.Blobs and Microsoft.Azure.Storage.Blob

Mohit-Chakraborty committed:
Update default doc index time to 10 seconds (#17914)

Mark the test inconclusive if indexing does not start.

azure-sdk committed:
Increment version for storage releases (#17916)

chamons committed:
[Synapse] Bump Swagger for nullability and naming fix (#17885)

- Pick up https://github.com/Azure/azure-rest-api-specs/pull/12258 and https://github.com/Azure/azure-rest-api-specs/pull/12390

Sandido committed:
Remove App Gateway tests (#17915)

* Delete VMScaleSetAppGWTests.cs

* Delete VMScaleSetAppGwWithAS.json

* Delete VMScaleSetAppGwWithAS_A1V2.json

* Delete VMScaleSetAppGwWithAS_Spot.json

* Delete VMScaleSetAppGwWithAS_SPG_False.json

dhawal777 committed:
Updating same version with additional changes. (#17854)

* Some additional changes in same version.

* Message correction.

* Reframed failed assert.

* Updating session record.

azure-sdk committed:
Update pipeline generation tool to support test variant pipelines (#17917)

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>

JoshLove-msft committed:
SB release (#17922)

* SB release

* reset apicompat version

jsquire committed:
[Docs] Contributing Guide Link Fix (#17919)

# Summary

The focus of these changes is to remove the locale from the links for PowerShell and Node installation.

# Last Upstream Rebase

Tuesday, December 12, 4:41pm (EST)

azure-sdk committed:
Increment version for servicebus releases (#17923)

Increment package version after release of Azure.Messaging.ServiceBus

azabbasi committed:
(ADT-DP) Add diagnostic scopes to all DigialTwins APIs (#17921)

ajaygit158 committed:
Updates as per swagger changes to refer SourceResourceSettings Property to ResourceSettings instead of MoveResourcePropertiesSourceResourceSettings. (#17148)

Co-authored-by: Ajay Gupta <ajgupt@microsoft.com>

jsquire committed:
[Event Hubs Client] Test Reliability Tweaks (#17941)

The focus of these changes is to tweak the approach and/or timing on
tests that were observed to be intermittently failing during pipeline
runs.

vidyadharijami committed:
Adding PPG and AvZone properties for V2A and H2A (#17853)

JoshLove-msft committed:
Simplify CloudEvent ctors and add dataSerializationType parameter (#17892)

* Simplify CloudEvent ctors and add dataSerializationType parameter

* Use JsonElement

* Export API

* Make data param required

scbedd committed:
Update CodeOwners to Reflect New Track2 Package (#17928)

* update mediaservices codeowners

jsquire committed:
[Messaging Clients] Test Resource Cleanup (#17938)

The focus of these changes is to cleanup the test resources now that the
pipelines have been split and also to introduce another element into GUID
generation for the RBAC rule names.

In the current form, we occasionally see collisions in parallel runs that
require restarting to fix.

harneetvirk committed:
Triggering pipelines for machinelearningservices (#17955)

JoshLove-msft committed:
Move ServiceBus extension to Azure SDK repo (#17951)

https://github.com/Azure/azure-sdk-for-net/issues/17560
Moving the extensions from https://github.com/Azure/azure-functions-servicebus-extension with minimal changes necessary to compile and have tests passed. This will be helpful as a reference for future changes as we migrate to use the Track 2 library.

Notable issues that should be converted to GH issues:
- signing the sign.snk did not work as there was no private key, so I'm using the SDK repo snk - https://github.com/Azure/azure-sdk-for-net/issues/17958
- TestCommon package did not support net461 so I updated the RequiredTargetFrameworks for the test project to just be netcoreapp2.1 - https://github.com/Azure/azure-sdk-for-net/issues/17959

HarveyLink committed:
Add Track1 code validation for MGMT pipeline (#17210)

* Update pipeline generate code check for Track1

* Create RPMapping.json

* Update script to get RP/folder in readme and remove rpmapping file

* Update RP Mapping code

* Add pipeline alert

* Update archetype-sdk-mgmt.yml

* Update archetype-sdk-mgmt.yml

* Update archetype-sdk-mgmt.yml

* Test for step change

* Update archetype-sdk-mgmt.yml

* Update timeout and fix issue

* Update Track1CodeCheck.ps1

azure-sdk committed:
Increment version for mediaservices releases (#17960)

Increment package version after release of Azure.Media.Analytics.Edge

azure-sdk committed:
Increment version for servicebus releases (#17949)

Increment package version after release of Microsoft.Azure.ServiceBus

chamons committed:
[synapse] SparkBatchOperation should return if complete OR started (#17920)

- We were blocking for the batch job to be complete, which was not the expected behavior outside of tests.

poanchen committed:
replace with name field (#17824)

crtreasu-msft committed:
Added shared authentication source for Mixed Reality clients (#17926)

This change configures shared source code to be used by Mixed Reality client libraries to handle authentication.

A simple project with documentation was added to maintain the integrity of the shared source.

chidozieononiwu committed:
Remove TestPipeline parameter from non-template pipelines. Add condition for setting test pipeline to true (#17911)

benbp committed:
Add cross product matrix tooling (#17417)

* Initial commit of test matrix generator

* Add multiple matrix job template

* Use powershell task syntax for matrix template

* Use inline syntax for matrix powershell functions

* Rename test-matrix samples and scripts to job-matrix

* [job matrix] Enforce naming restrictions. Extend matrix template in sample

* Switch matrix config processing to follow json declaration order

* Remove locale from job-matrix README links

* Fixes for samples and template

minnieliu committed:
[Communication] - Common - Rename Id to UserId (#17983)

* Rename Id to UserId

* Update generates files

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

chamons committed:
[synapse] Fix Sample6_HelloWorldLinkedService sample (#17974)

Sandido committed:
ContainerService APIs have been deprecated, removing tests (#17845)

* Delete ContainerServiceOperationalTests.cs

* Delete ContainerServiceTestsBase.cs

* Delete ContainerServiceUpdateTests.cs

* Delete TestDCOSOperations.json

* Delete TestSwarmOperations.json

* Delete TestContainerServiceUpdateOperations.json

* put back track2 OperationalTests

* put back track2 testbase

* put back track2 UpdateTests

* Delete ContainerServiceOperationalTests.cs

delete track1

* Delete ContainerServiceUpdateTests.cs

delete track1

* Delete ContainerServiceTestsBase.cs

delete track1

Sandido committed:
Release LogAnalytic Throttle API changes (#17939)

* Update Microsoft.Azure.Management.Compute.csproj

update version and release notes

* Update AssemblyInfo.cs

pakrym committed:
Suppress ObjectDisposedException from SslStream during request cancellation (#17963)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17527

azure-sdk committed:
Update AutoRest C# version (#17989)

pakrym committed:
Add project reference NET5 test configuration (#17967)

smile37773 committed:
Release for Azure Spring Cloud - 2020-11-01-preview (#17692)

* generate file

* release notes and versions

* add test

* update test

* delete TestFramework

benbp committed:
Change duplicate matrix name in live test template (#18009)

Mohit-Chakraborty committed:
Add migration guide in search (#18008)

* Add migration guide in search

The important link in the guide is to the Azure search upgrade doc

* Remove the lang part of the link

guomanmin committed:
added error code and removed all error messages (#12967)

* added error code and removed all error messages as discussed

* added error description back

* updated IncompatibleVMSize descirption and UnsupportedVMSize. Removed VMSizeNotSupportedByDesciption as it is not a PoolResizeErrorCode returned as Code (returned as Details.Code instead)

* fix casing for AllocationTimeout

Co-authored-by: maguo <maguo@microsoft.com>

maririos committed:
no check error code (#18011)

benbp committed:
Run smoke tests against latest published versions in release pipeline (#17524)

* Run smoke tests against latest published versions in release pipeline

* Update AzureAuthorityHosts reference in smoke tests

* Use env variable for RunSmokeTests pass through

* Only smoke test on release during manual build run

* Update smoke test eventhubs* dependency to 5.3.0-beta.*

* Smoke test: Use daily instead of nightly switch. Use only devops feed

* [smoke tests] Remove package exclusions. Only include GA/beta packages for daily==false

* Use latest non-alpha versions for smoke test csproj defaults

* Use build artifact version in smoke test after release

* Minor fixes for PR comments

* Bump smoke test dependency

* Add DoNotUpdate attribute for smoke test dependency

* [smoke test] Move build artifact version override logic into Update-Dependencies script

* [smoke test] Fix daily override conditional and bump core version

* Fix azure authority host references

* path fixes

* Use ubuntu 18 vmimage for smoke test eligibility check job

* Add net50 to smoke test matrix

kinelski committed:
[MetricsAdvisor] Ignore flaky tests (#18005)

jsquire committed:
[Event Hubs Client] Internalize Idempotent Publishing (#18016)

The focus of these changes is to internalize types and members related
to the idemtpotent publishing feature, as this will not be included in the
next Event Hubs release.

A correction was also made to the initial value of the last published
sequence number in the `PartitionPublishingProperties` to align with
service semantics.

Commit:
[Identity] update test.yml to use the updated ubuntu keyring container (#17985)

* [Identity] update test.yml to use the updated ubuntu keyring container

* updating to latest image

* update container again

ramya-rao-a committed:
Update Service Bus migration guide to follow template (#18019)

Resolves https://github.com/azure/azure-sdk-for-net/issues/16742 by updating the Service Bus migration guide to follow the template in https://github.com/Azure/azure-sdk/pull/2132

suhas92 committed:
[Text Analytics] Expose StringIndextype for all endpoints (#17968)

Expose stringindextype to all endpoints

jsquire committed:
[Event Hubs Client] Custom Endpoint Support (#18014)

The focus of these changes is to add an option for specifying a custom
address to use for establishing the connection to the Event Hubs service;
the namespace address will continue to be used for communication with the
service once the connection is established when the custom endpoint is
configured.

LarryOsterman committed:
Updated release date (#18020)

aarondandy committed:
Fixes ServiceBusProcessor.DisposeAsync exceptions (#18027)

This fixes an issues when a ServiceBusProcessor is disposed after it has been closed or is disposed multiple times. If a processor is used within a using block and also has Close explicitly called before the processor goes out of scope, it will throw an NRE.

Regarding multiple calls to dispose, I can't think of a good use case for why somebody would do that but it is documented behavior for IAsyncDisposable.

https://docs.microsoft.com/en-us/dotnet/api/system.iasyncdisposable.disposeasync?view=dotnet-plat-ext-5.0#remarks
> If an object's DisposeAsync method is called more than once, the object must ignore all calls after the first one and synchronously return a successfully completed ValueTask. The object must not throw an exception if its DisposeAsync method is called multiple times.

jsquire committed:
[Messaging Clients] Test Resource RBAC Grants (#18026)

The focus of these changes is to try and eliminate the potential for
duplicate role assignments during retries of template deployment.

The previous approach attempted to randomize the name sufficiently to avoid
collisions during retries; however, this increased failures because the
same grant cannot exist in the same scope for the same principal - regardless
of name.

This revision attempts to create a deterministic name based on the resource
and principal that the grant is associated with, and also to avoid performing
the assignment unless the key resources of the template were successful.

pakrym committed:
Support If-Modified-Since header (#18042)

christothes committed:
Fix partition key and row key sas token Uri strings (#18045)

fixes #17998

RezaJooyandeh committed:
Communication - Add Support for AzureKeyCredential (#18037)

* Communication - Add Support for AzureKeyCredential

* Update the variables in the recorded tests

* Fix some of the tests and remove unnessary wait for operations in playback mode

* Update the recorded tests for Sms client

* Update the recorded tests and changelogs

jsquire committed:
[Event Hubs Client] Minor Fixes (#18054)

The focus of these changes is to hide previously overlooked members related
to the idempotent publishing feature and to fix an inadvertent breaking change.

kinelski committed:
[MetricsAdvisor] Added AAD support and tests (#18036)

jsquire committed:
[Event Hubs Client] Preserve AMQP Exceptions (#18048)

The focus of these changes is to preserve AMQP exceptions as inner
exceptions after translating, to allow for deeper insight of service
behavior and assist in debugging efforts.

mikeharder committed:
[Perf] Add Storage GetBlobs test (#18053)

pakrym committed:
Disable default HttpClient/HttpWebRequest timeouts (#18050)

pakrym committed:
Skip empty checkpoints (#18059)

chidozieononiwu committed:
Add Language Display name to Language Settings file (#18056)

pakrym committed:
Avoid first chance exception on the hot path of RetriableStream (#18064)

tg-msft committed:
Storage: Coalesce prefix response headers into a single loop (#18021)

* Storage: Coalesce prefix response headers into a single loop

* Manually removed extra features from codegen

Generator changes are also getting merged, so any other branches merging
this in will get the changes regardless, but they also know exactly what
swagger they want to target.

* Cleanup previous commit

* Fixed build error; incorrectly deleted property.

* Accidentally removed a header read. Fixed.

* Reverted some FileRestClient renames`

Co-authored-by: jschrepp-MSFT <41338290+jschrepp-MSFT@users.noreply.github.com>

RezaJooyandeh committed:
[Communication] - Add identifier serializer (#18052)

* Communication - Add identifier serializer

* Add UTs

* Rename PhoneNumberIdentifier.Value to PhoneNumberIdentifier.PhoneNumber

amnguye committed:
Aligned storage URL parsing with other platforms (#18029)

azure-sdk committed:
Update error message in tools repo (#18067)

Co-authored-by: Chidozie Ononiwu <chononiw@microsoft.com>

JoshLove-msft committed:
Fix batching calculation (#18062)

alluri02 committed:
confluent updated dotnet SDK (#17997)

jsquire committed:
[Service Bus Client] Documentation Updates (#18063)

The focus of these changes is to update the changelog with a recent
community contribution and to reformat doc comments to try to fix a
doc generation issue.

azure-sdk committed:
Increment version for attestation releases (#18069)

Increment package version after release of Azure.Security.Attestation

chamons committed:
[synapse] Bump swagger to 6636b93e2c9eb979336841289ce08d7c6043ea0e  (#18061)

- Picks up Azure/azure-rest-api-specs#12501
- Also picked up a few other random changes from client team as well

chamons committed:
Rename *LiveTest.cs -> *LiveTests.cs to be consistent across repository (#18043)

- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17987
- I updated a handful of tests (4) outside of the template to match the majority (78) along with the template.

christothes committed:
Add transactional batch sample and clean up others (#18065)

* batch samples

mikeharder committed:
[Perf] Include running average in per-second output (#18071)

scbedd committed:
Flatten Docs Readme Writes (#17862)

* .NET cannot handle multiple overview pages. need to flatten down to a single folder on output. to do this we're taking advantage of a new suffix option in the CIConfigs variable

pakrym committed:
Include headers in the ExtractFailureContent partial methods (#18085)

Commit:
Deleted Obselete exception and added a scenario test to check for it (#17558)

* Deleted Obselete exception and added a scenario test

* bump up version

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#18092)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1317 (#18094)

* retarget the tool clone to a release tag

Co-authored-by: scbedd <45376673+scbedd@users.noreply.github.com>

HarveyLink committed:
Update Track1CodeCheck.ps1 for parsing and output (#18073)

* Update Track1CodeCheck.ps1

* Update Track1CodeCheck.ps1

* Update Track1CodeCheck.ps1

azure-sdk committed:
Ensure bypass conditional template gets run for 1es ubuntu image (#18097)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

mitchdenny committed:
Enable 1ES pools for live tests. (#17973)

* Enable 1ES pools for live tests.

alrod committed:
Update CHANGELOG.md (#18098)

vidyadharijami committed:
Supporting Availability set as input for Vmware to Azure and Hyper V to Azure scenarios (#18034)

HarveyLink committed:
Update script to have PR check (#18102)

HarveyLink committed:
Update Track1 MGMT pipeline (#18104)

* Update script to have PR check

* Update pipeline myl

* Update archetype-sdk-mgmt.yml

* Update archetype-sdk-mgmt.yml

HarveyLink committed:
Update Track1 code check script (#18107)

* Update script to have PR check

* Update pipeline myl

* Update archetype-sdk-mgmt.yml

* Update archetype-sdk-mgmt.yml

* Update Track1CodeCheck.ps1

kinelski committed:
[MetricsAdvisor] Enabling tests after service bug has been fixed (#18086)

chamons committed:
[Synapse] Rework and extend unit tests (#18010)

This is the first in a set of PRs intended to complete a basic level of test coverage on Synapse. This one in particular is more extension, and less additive, as it refactors and restructures the existing tests. 

Some notes:
- I've removed the shared testing base class, and moved some test helpers into static methods that declare the needed resources instead of digging into the base class's public variables. I also explicitly instance the test class in each test.
   - I prefer less magic and more explicit behavior as a rule. I find this easier to maintain.
   - I also updated test and sample namespaces to match the guidance from the template Foo.Bar.Tests or Foo.Bar.Samples
- A few new tests are included, just to prevent code coverage regressions. Some of them are [Ignored] due to current limitations in the test-resources.json instance, or issues with the service/SDK that I'm looking at.
- I am aware that some of the tests I fixed up in the refactor don't assert much, or could be improved further. As I noted above, additional test work will continue after this.
- I added a bunch of "stub" test classes just to fill out the set of test files, and to make it explicit what isn't being tested yet.

chamons committed:
[syanpse] Bump swagger to b9fbfae938a0ec0e8a696b450a917b47bea1f2e3 (#18089)

- Bump to include https://github.com/Azure/azure-rest-api-specs/pull/12518

weshaggard committed:
Clean build warnings and nuget config (#18095)

amnguye committed:
Regenerated for Azure.Storage.Files.Shares (#18072)

azure-sdk committed:
Update AutoRest C# version (#18022)

Update AutoRest C# version to 3.0.0-beta.20210115.1

abatishchev committed:
Used string.IsNullOrEmpty() in EnvironmentCredential (#18032)

* Used string.IsNullOrEmpty() in EnvironmentCredential

* Update EnvironmentCredential.cs

* Update CHANGELOG.md

* Update CHANGELOG.md

* Update CHANGELOG.md

* adding tests for environment variable empty strings

* reversing GetNonEmptyStringOrNull condition for readability

Co-authored-by: Scott Schaab <sschaab@microsoft.com>

heaths committed:
Remove secure key release implementation (#18100)

* Remove secure key release implementation

Resolves #17901

* Update public APIs

crtreasu-msft committed:
Updating the Mixed Reality test location (#18113)

Azure Spatial Anchors doesn't currently support West US 2, so the tests fail when run against that region. This change configures the test location to be a specific location (`eastus2`). This should resolve #17715.

azure-sdk committed:
Update AutoRest C# version (#18112)

aim-for-better committed:
Add some properties in some classes (#18075)

Co-authored-by: Zhenyu Zhou <zhezhou@microsoft.com>

harneetvirk committed:
Bootstrap sdk net for machinelearningservices (#17953)

* Add MachineLearningServices track 2 NET SDK

heaths committed:
Log x-ms-request-id for tests only (#18119)

Helps follow up with the service team if any service issues occur during nightly live test runs.

kasobol-msft committed:
[Storage] Capture azurite debug log before failing permanently. (#18117)

* capture azurite debug log before failing permanently.

* using...

* pr feedback.

christothes committed:
regen the client (#18126)

* regen the client

Sandido committed:
Throttle API log analytics release (#18123)

* Update Microsoft.Azure.Management.Compute.csproj

back to 42

* Update AssemblyInfo.cs

Commit:
Update SecurityInsights SDK [preview2] (#17969)

* Added teams to sdk from API

-Regenerated code using lastest API.
-Updated strings in tests
-Recoreded new test sessionrecords

* Ran generate.ps1

* Update tests due to api changes

* Version Update

Updated assembly versions and release notes

sima-zhu committed:
Added credscan steps for .NET (#17944)

chamons committed:
[codegen] Update to renamed code generator (#18122)

- Code generator renamed to include prefix https://github.com/Azure/autorest.csharp/issues/991

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1251 (#18124)

* Add PrepareRelease Script

* Generalize Prepare-Release Script

* Update Update-ChangeLog.ps1

* Update Package-Properties.ps1

* Update Collect-ChangeLog Script

* Update Collect-ChangeLog.ps1 and Collect-Unreleased.ps1

* Update GeneralReleaseNotesParser.ps1 script

* Update Prepare-Release Script to use recent DevOps Scripts

* Move Get-LanguageName to eng\common\scripts\Helpers\DevOps-WorkItem-Helpers.ps1, improve error message, add logic to detect changes made by prepare-release script

* Use LanguageDisplayName, move Get-CSVMetadata to Package-Properties.ps1

Co-authored-by: Chidozie Ononiwu <chononiw@microsoft.com>

maririos committed:
update swagger reference (#18128)

mikeharder committed:
[Perf] Show running average with 2 decimal places (#18134)

chamons committed:
[synapse] Add more Artifacts tests (#18114)

- IntegrationRuntimesClientLiveTests.cs 
- LinkedServiceClientLiveTests.cs 
- NotebookClientLiveTests.cs 
- PipelineClientLiveTests.cs 
- PipelineRunClientLiveTests.cs 
- SparkJobDefinitionClientLiveTests.cs

SimonCropp committed:
direct return from DeserializeAsync (#18129)

pakrym committed:
Switch to greedy strategy (#18140)

azure-sdk committed:
Update AutoRest C# version (#18125)

Update AutoRest C# version to 3.0.0-beta.20210121.3

RezaJooyandeh committed:
[Communication] - Identifier serialization new schema (#18120)

* Identity Derialization Requires Id

* Add cloud environment to MicrosoftTeamsIdentifier

* Update description

Co-authored-by: Dominik <domessin@microsoft.com>

Commit:
WIP

Commit:
Adding recorded files

Commit:
Resolve merge conflicts

RezaJooyandeh committed:
[Communication] - Make AsyncTokenRefresher in CommunicationTokenRefreshOptions optional (#18149)

* Communication - Make AsyncTokenRefresher in CommunicationTokenRefreshOptions optional

* Update the tests to do not have custom logic in creation of token credential

ashvermamsft committed:
Add New CloudService Resource in Azure SDK (#17341)

* adding cloudservice generated sdk files and test

* adding customizations to resolve runtime anbiguity

* regenrating the sdk from rest repo

* regenerating full SDK (including gallery and diskrp changes), fixing diskrp and gallery tests

* regenrating SDK using rest repo

Commit:
Resolve merge conflicts

Commit:
Merge

0rland0Wats0n committed:
[communication] fix endpoint env var name (#18138)

* fix endpoint env var name

* add COMMUNICATION_ACCESS_KEY env var

* change key1 to primaryKey

Commit:
Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net into managedIdentity

Commit:
Adding recordings for samples

Commit:
Adding recordings for samples

Commit:
Revert "Adding recordings for samples"

This reverts commit 753d1d5fd4b97c60878258639333a61afd68c819.

Commit:
Revert "Adding recordings for samples"

This reverts commit 753d1d5fd4b97c60878258639333a61afd68c819.

Commit:
Revert "Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net into managedIdentity"

This reverts commit 016c6113d66f534276e1257c2f31d1a6934ab9c5, reversing
changes made to bdb1731731b5a86a6d313f87bf5e66b0af5958fd.

Commit:
Revert "Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net into managedIdentity"

This reverts commit 016c6113d66f534276e1257c2f31d1a6934ab9c5, reversing
changes made to bdb1731731b5a86a6d313f87bf5e66b0af5958fd.

Commit:
Merge
